x-pack/plugins/fleet/public/applications/fleet/sections/agents/uninstall_token_list_page/index.tsx

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CriteriaWithPagination, EuiBasicTableColumn } from '@elastic/eui';
+import { EuiFieldSearch } from '@elastic/eui';
+import { EuiToolTip } from '@elastic/eui';
+import { EuiButtonIcon } from '@elastic/eui';
+import { EuiSpacer } from '@elastic/eui';
+import { EuiBasicTable, EuiText } from '@elastic/eui';
+import React, { useCallback, useMemo, useState } from 'react';
+import { FormattedDate, FormattedMessage } from '@kbn/i18n-react';
+import type { SendRequestResponse } from '@kbn/es-ui-shared-plugin/public';
+
+import { ApiKeyField } from '../../../../../components/api_key_field';
+import type { UninstallTokenMetadata } from '../../../../../../common/types/models/uninstall_token';
+import {
+  sendGetUninstallToken,
+  useGetUninstallTokens,
+} from '../../../../../hooks/use_request/uninstall_tokens';
+import { useBreadcrumbs, usePagination } from '../../../hooks';
+import { DefaultLayout } from '../../../layouts';
+import type { GetUninstallTokenResponse } from '../../../../../../common/types/rest_spec/uninstall_token';
+import { UninstallCommandFlyout } from '../../../components';
+
+import {
+  ACTIONS_TITLE,
+  CREATED_AT_TITLE,
+  GET_UNINSTALL_COMMAND_LABEL,
+  POLICY_ID_TITLE,
+  SEARCH_BY_POLICY_ID_PLACEHOLDER,
+  TOKEN_TITLE,
+} from './translations';
+
+const PolicyIdField = ({ policyId }: { policyId: string }) => (
+  <EuiText
+    size="s"
+    className="eui-textTruncate"
+    title={policyId}
+    data-test-subj="uninstallTokensPolicyIdField"
+  >
+    {policyId}
+  </EuiText>
+);
+
+const GetUninstallCommandButton = ({ onClick }: { onClick: () => void }) => (
+  <EuiToolTip content={GET_UNINSTALL_COMMAND_LABEL}>
+    <EuiButtonIcon
+      data-test-subj="uninstallTokensUninstallButton"
+      aria-label={GET_UNINSTALL_COMMAND_LABEL}
+      onClick={onClick}
+      iconType="minusInCircle"
+      color="danger"
+    />
+  </EuiToolTip>
+);
+
+const NoItemsMessage = ({ isLoading }: { isLoading: boolean }) =>
+  isLoading ? (
+    <FormattedMessage
+      id="xpack.fleet.uninstallTokenList.loadingTokensMessage"
+      defaultMessage="Loading uninstall tokens..."
+    />
+  ) : (
+    <FormattedMessage
+      id="xpack.fleet.uninstallTokenList.emptyMessage"
+      defaultMessage="No uninstall tokens found."
+    />
+  );
+
+export const UninstallTokenListPage = () => {
+  useBreadcrumbs('uninstall_tokens');
+
+  const [policyIdSearch, setPolicyIdSearch] = useState<string>('');
+  const [tokenIdForFlyout, setTokenIdForFlyout] = useState<string | null>(null);
+
+  const { pagination, setPagination, pageSizeOptions } = usePagination();
+
+  const { isLoading, data } = useGetUninstallTokens({
```

Comment:
we try to move to react query in Fleet, as it's a new page, do you mind replacing that call by a react query one?  should be something like
```
  const { data: healthCheckResponse } = useQuery(
    ['getUninstallTokens'],
    () => sendGetUninstallTokens(),
```

Reply:
sure! :)
8456dc88e3c2122e6c50a237606b43047476cd47